### PR TITLE
recreate all features from deprecated code and remove that old code

### PR DESCRIPTION
**Pull request by AmsTaFFix**

### CODE REVIEW

File: src/CsvReader.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            // convert to assoc array
+            $assocRow = [];
+            foreach ($columnNames as $index => $columnName) {
+                $assocRow[$columnName] = $row[$index];
+            }
+
+            yield $assocRow;
+        }
+    }
+}
```
Comment (liderman) [inline src/CsvReader.php]: Please add an empty string at the end of the file

File: src/Model/Column.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Get value
+     *
+     * @return string|null Value
+     */
+    public function getValue(): ?string
+    {
+        return $this->value;
+    }
+}
```
Comment (liderman) [inline src/Model/Column.php]: Please add an empty string at the end of the file

File: src/Model/DataTablesAddMultipleRowsPostRequest/Row.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     * Get columns
+     *
+     * @return Column[]|iterable Columns
+     */
+    public function getColumns()
+    {
+        return $this->columns;
+    }
+
+}
```
Comment (liderman) [inline src/Model/DataTablesAddMultipleRowsPostRequest/Row.php]: Please add an empty string at the end of the file

File: src/Model/DataTablesGetDataPostRequest/OrderByRule.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public function __construct(string $columnName, Direction $direction)
+    {
+        Assert::notEmpty($columnName);
+        $this->columnName = $columnName;
+        $this->direction = $direction;
+    }
+
+    /**
+     * @return string
```
Comment (liderman) [inline src/Model/DataTablesGetDataPostRequest/OrderByRule.php]: Please add documentation

File: src/Model/DataTablesGetDataPostRequest/OrderByRule.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public function getColumnName(): string
+    {
+        return $this->columnName;
+    }
+
+    /**
+     * @return Direction
```
Comment (liderman) [inline src/Model/DataTablesGetDataPostRequest/OrderByRule.php]: Please add documentation

File: src/Model/DataTablesGetDataPostRequest/WhereCondition.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     * Get value
+     *
+     * @return string Value
+     */
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+}
```
Comment (liderman) [inline src/Model/DataTablesGetDataPostRequest/WhereCondition.php]: Please add an empty string at the end of the file

File: src/Request/DataTablesDeleteRowPostRequest.php [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getUri(): string
+    {
+        return '/Api/DataTablesDeleteRow';
+    }
+}
```
Comment (liderman) [inline src/Request/DataTablesDeleteRowPostRequest.php]: Please add an empty string at the end of the file

File: src/Request/DataTablesGetDataPostRequest.php [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getUri(): string
+    {
+        return '/Api/DataTablesGetData';
+    }
+}
```
Comment (liderman) [inline src/Request/DataTablesGetDataPostRequest.php]: Please add an empty string at the end of the file

File: src/Request/DataTablesUpdateRowPostRequest.php [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getUri(): string
+    {
+        return '/Api/DataTablesUpdateRow';
+    }
+}
```
Comment (liderman) [inline src/Request/DataTablesUpdateRowPostRequest.php]: Please add an empty string at the end of the file

File: src/Traits/ColumnToXmlConverterTrait.php [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+            $xmlWriter->startElement('Value');
+            $xmlWriter->writeAttributeNS('xsi', 'nil', null, 'true');
+            $xmlWriter->endElement(); // Value
+        } else {
+            $xmlWriter->writeElement('Value', $column->getValue());
+        }
+
+        $xmlWriter->endElement(); // Column
+    }
+}
```
Comment (liderman) [inline src/Traits/ColumnToXmlConverterTrait.php]: Please add an empty string at the end of the file

File: src/CsvReader.php (hunk 1 above)
Comment (AmsTaFFix) [inline src/CsvReader.php]: added

File: src/Model/Column.php (hunk 2 above)
Comment (AmsTaFFix) [inline src/Model/Column.php]: added

File: src/Model/DataTablesAddMultipleRowsPostRequest/Row.php (hunk 3 above)
Comment (AmsTaFFix) [inline src/Model/DataTablesAddMultipleRowsPostRequest/Row.php]: added

File: src/Request/DataTablesDeleteRowPostRequest.php (hunk 7 above)
Comment (AmsTaFFix) [inline src/Request/DataTablesDeleteRowPostRequest.php]: added

File: src/Request/DataTablesGetDataPostRequest.php (hunk 8 above)
Comment (AmsTaFFix) [inline src/Request/DataTablesGetDataPostRequest.php]: added

File: src/Request/DataTablesUpdateRowPostRequest.php (hunk 9 above)
Comment (AmsTaFFix) [inline src/Request/DataTablesUpdateRowPostRequest.php]: added

File: src/Traits/ColumnToXmlConverterTrait.php (hunk 10 above)
Comment (AmsTaFFix) [inline src/Traits/ColumnToXmlConverterTrait.php]: added

**Pull request merged**